File: tree/617.c
/**
 * recursive and recursive
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* mergeTrees(struct TreeNode* t1, struct TreeNode* t2) {
    if(t1 && t2) 
    {
        struct TreeNode* root = (struct TreeNode*)malloc(sizeof(struct TreeNode));
        root->val = t1->val + t2->val;
        root->left = mergeTrees(t1->left, t2->left);
        root->right = mergeTrees(t1->right, t2->right);
        return root;
    }
    if(t1 && !t2) return t1;
    if(!t1 && t2) return t2;
    return NULL;
    
}

File: tree/617.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func mergeTrees(t1 *TreeNode, t2 *TreeNode) *TreeNode {
    if t1 != nil && t2 != nil {
        newNode := &TreeNode{0, nil, nil}
        newNode.Val = t1.Val + t2.Val
        newNode.Left = mergeTrees(t1.Left, t2.Left)
        newNode.Right = mergeTrees(t1.Right, t2.Right)            
        return newNode
    }
    if t1 == nil {
        return t2
        // newNode.Val = t2.Val
        // newNode.Left = mergeTrees(nil, t2.Left)
        // newNode.Right = mergeTrees(nil, t2.Right)
    }
    if t2 == nil {
        return t1
        // newNode.Val = t1.Val
        // newNode.Left = mergeTrees(nil, t1.Left)
        // newNode.Right = mergeTrees(nil, t1.Right)        
    }
    return nil
}
